lambda_functions/v1/functions/lpa_codes/app/lpa_codes_mock.py

```diff
@@ -0,0 +1,127 @@
+import connexion
+import boto3
+import os
+from flask import request, jsonify
+from connexion.exceptions import OAuthProblem
+
+TOKEN_DB = {"asdf1234567890": {"uid": 100}}
+
+
+def apikey_auth(token, required_scopes):
+    info = TOKEN_DB.get(token, None)
+
+    if not info:
+        raise OAuthProblem("Invalid token")
+
+    return info
+
+
+def aws_credentials():
+    """Mocked AWS Credentials for moto."""
+    os.environ["AWS_ACCESS_KEY_ID"] = "testing"
+    os.environ["AWS_SECRET_ACCESS_KEY"] = "testing"
+    os.environ["AWS_SECURITY_TOKEN"] = "testing"
+    os.environ["AWS_SESSION_TOKEN"] = "testing"
+    os.environ["AWS_DEFAULT_REGION"] = "eu-west-1"
+
+
+def setup_dynamodb_table():
+    aws_credentials()
+    ddb = boto3.client(
+        "dynamodb", endpoint_url="http://localhost:8000", region_name="eu-west-1"
+    )
+    try:
+        table = ddb.create_table(
+            TableName="lpa_codes",
+            KeySchema=[{"AttributeName": "code", "KeyType": "HASH"}],  # Partition key
+            AttributeDefinitions=[{"AttributeName": "code", "AttributeType": "S"}],
+            ProvisionedThroughput={"ReadCapacityUnits": 10, "WriteCapacityUnits": 10},
+        )
+        key_index = [
+            {
+                "Create": {
+                    "IndexName": "key_index",
+                    "KeySchema": [
+                        {"AttributeName": "lpa", "KeyType": "HASH"},
+                        {"AttributeName": "actor", "KeyType": "RANGE"},
+                    ],
+                    "Projection": {"ProjectionType": "ALL"},
+                    "ProvisionedThroughput": {
+                        "ReadCapacityUnits": 5,
+                        "WriteCapacityUnits": 5,
+                    },
+                }
+            },
+        ]
+
+        active_index = [
+            {
+                "Create": {
+                    "IndexName": "active_index",
+                    "KeySchema": [{"AttributeName": "active", "KeyType": "HASH"}],
+                    "Projection": {"ProjectionType": "ALL"},
+                    "ProvisionedThroughput": {
+                        "ReadCapacityUnits": 5,
+                        "WriteCapacityUnits": 5,
+                    },
+                }
+            },
+        ]
+
+        table.update(GlobalSecondaryIndexUpdates=key_index)
+        table.update(GlobalSecondaryIndexUpdates=active_index)
+
+    except ddb.exceptions.ResourceInUseException:
+        print("Table already exists")
+
+
+def update_state():
+    aws_credentials()
+    mapping = {
+        "generated code exists and active": setup_code_active,
+        "generated code exists and not active": setup_code_not_active,
+    }
+    mapping[request.json["state"]]()
+    return jsonify({"result": request.json["state"]})
+
+
+def setup_code_active():
```

Comment:
You are using a mixture of `client` and `resource` to access dynamodb. Not sure which is right tbh, I am currently using resource but was about to switch to client because it as more useful methods 🤷